tests/std/tests/P0896R4_views_transform/test.cpp

```diff
@@ -321,6 +321,50 @@ constexpr bool test_one(Rng&& rng, Expected&& expected) {
     return true;
 }
 
+// Test a function object whose const and non-const versions behave differently
+struct difference_teller {
+    constexpr auto& operator()(auto&& x) noexcept {
+        auto& ref = x;
+        return ref;
+    }
+
+    constexpr auto operator()(auto&& x) const noexcept {
+        return type_identity<decltype(x)>{};
+    }
+};
+
+template <ranges::input_range Rng>

```

Comment:
This feels like a weird test since it seems to depend on the properties of difference_teller, but also kind of not? I think it'd be better to do:

```cxx
template <ranges::input_range<Rng>>
    requires same_as<ranges::range_value_t<Rng>, difference_teller>
```

Reply:
I failed to understand the intent of the suggestion, since the test file is definitely not testing any range whose value type is `difference_teller`, and I believe `difference_teller` works with all input ranges...

The test is somehow unbalanced to me as it test a large number of ranges but only a small number of function object types.

Reply:
oh, my brain absolutely broke, I'm sorry. nvm.